src/Core/Parser/SyntaxTree/AbstractNode.php

```diff
@@ -31,20 +31,18 @@
      */
     public function evaluateChildNodes(RenderingContextInterface $renderingContext)
     {
-        $childNodes = $this->getChildNodes();
-        $numberOfChildNodes = count($childNodes);
-        if ($numberOfChildNodes === 0) {
-            return null;
+        $evaluatedNodes = [];
+        foreach ($this->getChildNodes() as $childNode) {
+            $evaluatedNodes[] = $this->evaluateChildNode($childNode, $renderingContext, false);
         }
-        if ($numberOfChildNodes === 1) {
-            return $this->evaluateChildNode($childNodes[0], $renderingContext, false);
+        // Make decisions about what to actually return
+        if (empty($evaluatedNodes)) {
+            return null;
         }
-        $output = '';
-        /** @var $subNode NodeInterface */
-        foreach ($childNodes as $subNode) {
-            $output .= $this->evaluateChildNode($subNode, $renderingContext, true);
+        if (count($evaluatedNodes) === 1) {
+            return $evaluatedNodes[0];
         }
-        return $output;
+        return (string) implode('', array_map([$this, 'cast'], $evaluatedNodes));
```

Comment:
You can drop the string cast for the return value of `implode()` since that function always returns a string.